File: app/src/main/java/ru/vegax/xavier/a3test/adapters/AlbumsAdapter.java
package ru.vegax.xavier.a3test.adapters;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.ArrayList;
import java.util.List;

import ru.vegax.xavier.a3test.R;
import ru.vegax.xavier.a3test.models.UserAlbum;


public abstract class AlbumsAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> implements View.OnClickListener{

    private List<UserAlbum> mUserAlbums;
    private Context mContext;
    private Integer mUserId;
    private List<UserAlbum> mCurrentAlbums;

    protected AlbumsAdapter(@NonNull Context context) {
        mContext = context;
    }

    @Override
    public int getItemCount() {


        if (mCurrentAlbums != null){

        return mCurrentAlbums.size();
        }
        return 0;
    }


    @NonNull
    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {

        return new ViewHolderUserAlbums(LayoutInflater.from(mContext).inflate(R.layout.list_item_user, parent, false));


    }


    @Override
    public void onBindViewHolder(@NonNull RecyclerView.ViewHolder holder, int position) {

        ViewHolderUserAlbums viewHolderUsers = (ViewHolderUserAlbums) holder;
        //Get current item
        UserAlbum currentItem = mCurrentAlbums.get(position);

        viewHolderUsers.bindTo(currentItem);


        TextView textView = viewHolderUsers.getTextView();
        //disable swipe for the switch, only click on the item will trigger the output

        if (currentItem != null) {
            textView.setTag(currentItem.getId()); // album Id
            textView.setOnClickListener(this);
        }
    }

    public void setUserList(List<UserAlbum> userAlbums, Integer userId) {
        mUserAlbums = userAlbums;
        mUserId = userId;
        mCurrentAlbums = new ArrayList<>();
        for (UserAlbum userAlbum:mUserAlbums) {
            if (userAlbum.getUserId() == mUserId){
                mCurrentAlbums.add(userAlbum);
            }
        }
        notifyDataSetChanged();
    }

    class ViewHolderUserAlbums extends RecyclerView.ViewHolder {

        private TextView mTxtVUser;

        TextView getTextView() {
            return mTxtVUser;
        }


        ViewHolderUserAlbums(View itemView) {
            super(itemView);
            //Initialize the views
            mTxtVUser = itemView.findViewById(R.id.txtVUser);
        }

        void bindTo(UserAlbum currentItem) {
            if (currentItem!= null){
            mTxtVUser.setText(currentItem.getTitle());
            }
        }
    }
}

File: app/src/main/java/ru/vegax/xavier/a3test/adapters/AlbumsAdapter.kt
package ru.vegax.xavier.a3test.adapters

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView

import java.util.ArrayList

import ru.vegax.xavier.a3test.R
import ru.vegax.xavier.a3test.models.UserAlbum


abstract class AlbumsAdapter protected constructor(private val mContext: Context) : RecyclerView.Adapter<RecyclerView.ViewHolder>(), View.OnClickListener {

    private var mUserAlbums: List<UserAlbum>? = null
    private var mUserId: Int? = null
    private var mCurrentAlbums: MutableList<UserAlbum>? = null

    override fun getItemCount(): Int {


        return if (mCurrentAlbums != null) {

            mCurrentAlbums!!.size
        } else 0
    }


    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {

        return ViewHolderUserAlbums(LayoutInflater.from(mContext).inflate(R.layout.list_item_user, parent, false))


    }


    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {

        val viewHolderUsers = holder as ViewHolderUserAlbums
        //Get current item
        val currentItem = mCurrentAlbums!![position]

        viewHolderUsers.bindTo(currentItem)


        val textView = viewHolderUsers.textView
        //disable swipe for the switch, only click on the item will trigger the output

        textView.tag = currentItem.id // album Id
        textView.setOnClickListener(this)
    }

    fun setUserList(userAlbums: List<UserAlbum>?, userId: Int?) {
        mUserAlbums = userAlbums
        mUserId = userId
        mCurrentAlbums = ArrayList()
        if (userAlbums != null) {
            for (userAlbum in mUserAlbums!!) {
                if (userAlbum.userId == mUserId) {
                    mCurrentAlbums!!.add(userAlbum)
                }
            }
        }

        notifyDataSetChanged()
    }

    internal inner class ViewHolderUserAlbums(itemView: View) : RecyclerView.ViewHolder(itemView) {

        val textView: TextView


        init {
            //Initialize the views
            textView = itemView.findViewById(R.id.txtVUser)
        }

        fun bindTo(currentItem: UserAlbum?) {
            if (currentItem != null) {
                textView.text = currentItem.title
            }
        }
    }
}
